spring-core/src/main/java/org/springframework/util/xml/DomContentHandler.java

```diff
@@ -50,8 +50,8 @@ class DomContentHandler implements ContentHandler {
 	 */
 	DomContentHandler(Node node) {
 		this.node = node;
-		if (node instanceof Document) {
-			this.document = (Document) node;
+		if (node instanceof Document document) {
```

Comment:
This causes the build to fail due to a Checkstyle bug.

I've addressed it locally on my machine and will create an issue to track the Checkstyle bug.